packages/squiggle-lang/src/fr/list.ts

```diff
@@ -192,6 +192,41 @@ export const library = [
       ),
     ],
   }),
+  maker.make({
+    name: "reduceWhile",
+    requiresNamespace: true,
+    examples: [
+      // Args: (list, initialValue, step, condition)
+      // Returns the last value that fits the condition.
+      // If even initial value doesn't fit the condition, it will be returned anyway;
+      // So the result isn't guaranteed to fit the condition.
+      `List.reduceWhile([1,4,5], 0, {|acc, curr| acc + curr }, {|acc| acc < 5})`,
+    ],
+    definitions: [
+      makeDefinition(
+        [frArray(frAny), frAny, frLambda, frLambda],
+        ([array, initialValue, step, condition], context) => {
+          let acc = initialValue;
+          for (let i = 0; i < array.length; i++) {
+            const newAcc = step.call([acc, array[i]], context);
```

Comment:
minor note, but I could see it being good for ``reduce`` and ``reduceWhile`` to also optionally give an ``index``. Can do later though.